### ISSUE TITLE
Glob imports ignore extern crate items of parent modules

### ISSUE THREAD

**Issue by crumblingstatue** [rust-lang/rust#22146]
It doesn't matter if the extern crate item is public or private, it's ignored in both cases.

Example:
```
#![feature(libc)]

pub mod foo {
    pub extern crate libc;
    pub mod bar {
        // use foo::libc; // Direct import works
        use foo::*; // Glob import doesn't work
        pub fn bar_fn() {
            let _: libc::c_int;
        }
    }
}

fn main() {
    foo::bar::bar_fn();
}
```
This gives the error:
```
<anon>:9:20: 9:31 error: failed to resolve. Use of undeclared type or module `libc`
<anon>:9             let _: libc::c_int;
                            ^~~~~~~~~~~
<anon>:9:20: 9:31 error: use of undeclared type name `libc::c_int`
<anon>:9             let _: libc::c_int;
                            ^~~~~~~~~~~
```

Note that this is not the same issue as #22145, as it doesn't matter whether the item is public or private.

**Comment by durka** [comment]
This seems to be fixed in nightly!

**Comment by jseyfried** [comment]
This was fixed in #31362, after which `extern crate`s are treated the same as non-import items.
cc @steveklabnik
